src/DeBruijnGraph.jl

```diff
@@ -0,0 +1,281 @@
+"""
+    DeBruijnGraph is also SequenceGraph with a special design concept
+
+"""
+
+struct DeBruijnGraph
+    nodes::Vector{SequenceGraphNode}
+    links::Vector{Vector{SequenceGraphLink}}
+end
+
+
+nodes(dbg::DeBruijnGraph) = dbg.nodes
+node(dbg::DeBruijnGraph, i::NodeID) = nodes(dbg)[abs(i)]
+links(dbg::DeBruijnGraph) = dbg.links
+indegree(dbg::DeBruijnGraph,i::NodeID) = count_indegree(dbg,i)
+
+
+
+"""
+    links(sg::SequenceGraph, node::NodeID)
+
+Get all of the links of a Node of a sequence graph.
+"""
+function links(dbg::DeBruijnGraph, node::NodeID)
+    l = links(dbg)
+    @inbounds return l[abs(node)]
+end
+
+"""
+
+For now this is empty. We have to decide about the design concepts first!!!
+"""
+function add_link(dbg::DeBruijnGraph,l::SequenceGraphLink)
+
+end
+
+"""
+    find_overlaps(X::Vector{SequenceGraphNode})
+
+Finds the overlaps of length length(Node.sequence)-1 between all Nodes in the vector X
+
+Returns a vector of Tuples where each Tuple (i,j) corresonds to a directed edge (forward link) from i to j
+
+For constructing the De Bruijn Graph links we need the overlapping sequence information
+We first formulate the forward links as tuples with indices
+Then using these tuple indices we will create the SequenceGraphLinks and finalize the Graph Construction
+"""
+function find_overlaps(X::Vector{SequenceGraphNode})
+    overlaps = Vector{Tuple{Int64,Int64}}()
+    for i in 1:size(X)[1]
+        for j in 1:size(X)[1]
```

Comment:
I used the .size function instead of length as I got some issues when using the length function when I include the .jl files of BioSequenceGraphs package directly.  

<img width="1115" alt="length_issue" src="https://user-images.githubusercontent.com/12223406/58600628-57ce9980-82c0-11e9-99d9-ecf4afb5df43.png">